nicegui/elements/progress.py

```diff
@@ -1,48 +1,124 @@
+        text_color: Optional[str] = "white",
+        dark: bool = False,
+        rounded: bool = False,
+        animation_speed: int = 2100,
+        indeterminate: bool = False,
+        query: bool = False,
+        instant_feedback: bool = False,
+        stripe: bool = False,
+    ) -> None:
         """Linear Progress
 
         A linear progress bar wrapping Quasar's
         `QLinearProgress <https://quasar.dev/vue-components/linear-progress>`_ component.
+        It provides a way to display a progress bar with customizable options.
 
-        :param value: the initial value of the field (from 0.0 to 1.0)
-        :param size: the height of the progress bar (default: "20px" with value label and "4px" without)
-        :param show_value: whether to show a value label in the center (default: `True`)
-        :param color: color (either a Quasar, Tailwind, or CSS color or `None`, default: "primary")
+        :param value: The initial value of the progress bar (from 0.0 to 1.0).
+        :param min: The minimum value of the progress bar (default: 0.0).
+        :param max: The maximum value of the progress bar (default: 1.0).
+        :param size: The height of the progress bar (default: None).
+        :param show_value: Whether to show a value label in the center (default: True).
+        :param color: The color of the progress bar (either a Quasar, Tailwind, or CSS color or `None`, default: "primary").
+        :param reverse: Whether to reverse the direction of the progress bar (default: False).
+        :param buffer: The buffer value of the progress bar (default: None).
+        :param track_color: The color of the progress bar track (either a Quasar, Tailwind, or CSS color or `None`, default: None).
+        :param dark: Whether to use dark mode for the progress bar (default: False).
+        :param rounded: Whether to use rounded corners for the progress bar (default: False).
+        :param animation_speed: The animation speed of the progress bar (default: 2100).
+        :param indeterminate: Whether the progress bar is indeterminate (default: False).
+        :param query: Whether the progress bar is in query mode (default: False).
+        :param instant_feedback: Whether to provide instant feedback for the progress bar (default: False).
+        :param stripe: Whether to show a stripe pattern on the progress bar (default: False).
         """
-        super().__init__(tag='q-linear-progress', value=value, on_value_change=None, text_color=color)
-        self._props['size'] = size if size is not None else '20px' if show_value else '4px'
+        if max > 1.0:
+            value = float(value / max)
+
+        super().__init__(
+            tag="q-linear-progress",
+            value=value,
+            on_value_change=None,
+            text_color=color,
+        )
+        self._props["min"] = min
+        self._props["max"] = max
+
+        if isinstance(size, int):
+            size = str(size) + "px"
```

Comment:
I'm not sure if I like the implicit `px` unit. Why not simply take `str` sizes like `ui.icon` does?

Reply:
There are two ways I thought we could use the `size` arg, one using it as a simple string like on `ui.icon` and some other places but we might also want to use it as an integer for fast prototyping, its easier to use a number than to figure out what string to use for the right size, so, the `px` unit is to make sure that when we use an integer we use pixels and not some other unit but this can be changes. We could also have an argument for the default `unit` to use to make it even more customizable.